server/server/src/billing/stripe.rs

```diff
@@ -0,0 +1,219 @@
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug, Serialize, Deserialize)]
+#[serde(untagged)]
+pub enum StripeResult<U> {
+    Ok(U),
+    Err(StripeErrorContainer),
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct StripeErrorContainer {
+    pub error: StripeError,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct StripeError {
+    #[serde(rename = "type")]
+    pub error_type: StripeErrorType,
+    pub code: StripeMaybeContainer<StripeKnownErrorCode, String>,
+    pub decline_code: Option<StripeMaybeContainer<StripeKnownErrorDeclineCode, String>>,
+    pub doc_url: String,
+    pub message: String,
+    pub param: String,
+    pub payment_method_type: Option<String>,
+    pub charge: Option<String>,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub enum StripeErrorType {
+    #[serde(rename = "api_error")]
+    ApiError,
+    #[serde(rename = "card_error")]
+    CardError,
+    #[serde(rename = "idempotency_error")]
+    IdempotencyError,
+    #[serde(rename = "invalid_request_error")]
+    InvalidRequestError,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
```

Comment:
I think you want to add more comments in genera, it's not clear to me what a `StripeMaybeContainer` is at all